### ISSUE TITLE
Submodule update doesn't work with relative URL's

### ISSUE THREAD

**Issue by davidebbo** [projectkudu/kudu#1883]
Issue comes from [forum](https://social.msdn.microsoft.com/Forums/azure/en-US/3d74f352-20a2-4ace-82dd-c444242154fd/azure-git-deploy-error-with-submodule-on-relative-url?forum=windowsazurewebsitespreview).

The test repo is https://github.com/gobelineKiller/MainRepo. It has this `.gitmodules` file:

```
[submodule "SubRepo"]
        path = SubRepo
        url = ../SubRepo.git
```

The expected semantic on the `..` is that it's relative to the repo URL. So it should point to https://github.com/gobelineKiller/SubRepo.

But today, this doesn't work in Kudu.

**Comment by davidebbo** [comment]
The reason it doesn't work has to do with the way Kudu deals with remotes in an ephemeral way. A remote called `external` is added just before doing the fetch operation, and is then removed right after. By the time we update the submodules, the remote is gone, so git is not able to interpret the relative url.

I don't remember the history behind the short lived `external` remote (it's been like that for years), and what bad things would happen if we didn't delete it. This is the first time it causes an issue, so the impact is low. But ideally we would address it in some way.

**Comment by davidebbo** [comment]
Fixed by #1888
